src/Cake/Arguments/ArgumentParser.cs

```diff
@@ -67,10 +66,11 @@ public CakeOptions Parse(IEnumerable<string> args)
                             // Make sure we parse the option
                             if (!ParseOption(value, options))
                             {
-                                return null;
+                                options.HasError = true;
+                                return options;
                             }
 
-                            options.Script = GetDefaultScript();
+                            options.Script = "./build.cake";
```

Comment:
@devlead @gep13 

Removed conventions we defined here since we only have one real convention for file names now. Also removes the responsibility of the tokenizer to determine whether or not files exist (which shouldn't be it's job).

Ok with you?